### PR TITLE
fix(frontend): moved AuthWrapper from protected route into Root component bb-258

### PR DESCRIPTION
**Pull request by ihorLysak**
### Summary
Fixed by moving authWrapper from ProtectedRoute component into individual page (currently Root component)
Screenshot 2024-09-02 at 16 26 48
Screenshot 2024-09-02 at 16 26 38

### CODE REVIEW

File: apps/frontend/src/pages/root/root.tsx [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { AuthWrapper } from "~/libs/components/components.js";
+
+const Root: React.FC = () => {
```
Comment (fshabanov) [inline apps/frontend/src/pages/root/root.tsx:3]: I don't like this solution. It means in future for any new screens we need to wrap in in AuthWrapper. We should keep AuthWrapper in ProtectedRoute and think of another way of dealing with the issue

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by marharita08 — APPROVED** [review]

File: apps/frontend/src/libs/components/protected-route/libs/constants/routes-with-auth-wrapper.constant.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { AppRoute } from "~/libs/enums/enums.js";
+
+const routesWithoutAuthWrapper = [AppRoute.QUIZ] as string[];
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/libs/constants/routes-with-auth-wrapper.constant.ts:3]:
```suggestion
const routesWithoutAuthWrapper = [AppRoute.QUIZ] as ValueOf<typeof AppRoute>[];
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/protected-route/libs/helpers/has-auth-wrapper.helper.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { type AppRoute } from "~/libs/enums/app-route.enum.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/libs/helpers/has-auth-wrapper.helper.ts:1]:
```suggestion
import { type AppRoute } from "~/libs/enums/enums.js";
```

File: apps/frontend/src/libs/components/protected-route/protected-route.tsx [hunk 4]
```diff
@@ -1,9 +1,10 @@
 import { Navigate } from "~/libs/components/components.js";
 import { type AppRoute } from "~/libs/enums/enums.js";
-import { useAppSelector } from "~/libs/hooks/hooks.js";
+import { useAppSelector, useLocation } from "~/libs/hooks/hooks.js";
 import { type ValueOf } from "~/libs/types/types.js";
 
 import { AuthWrapper } from "../auth-wrapper/auth-wrapper.js";
+import { hasAuthWrapper } from "./libs/helpers/has-auth-wrapper.helper.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/protected-route.tsx:7]:
```suggestion
import { hasAuthWrapper } from "./libs/helpers/helpers.js";
```

File: apps/frontend/src/libs/components/protected-route/protected-route.tsx [hunk 5]
```diff
@@ -16,10 +17,16 @@ const ProtectedRoute: React.FC<Properties> = ({
 }: Properties) => {
 	const user = useAppSelector(({ auth }) => auth.user);
 
-	if (user) {
+	const { pathname } = useLocation();
+
+	if (user && hasAuthWrapper(pathname as ValueOf<typeof AppRoute>)) {
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/protected-route.tsx:22]: Let's simplify:
```
if (!user) {
  return <Navigate ... />
}

if (hasAuthWrapper(...)) {
  return <AuthWrapper>{component}</AuthWrapper>
}

return component
```

File: apps/frontend/src/libs/components/protected-route/libs/constants/routes-with-auth-wrapper.constant.ts [hunk 6]
```diff
@@ -0,0 +1,6 @@
+import { AppRoute } from "~/libs/enums/enums.js";
+import { type ValueOf } from "~/libs/types/types.js";
+
+const routesWithoutAuthWrapper = [AppRoute.QUIZ] as ValueOf<typeof AppRoute>[];
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/libs/constants/routes-with-auth-wrapper.constant.ts:4]:
```suggestion
const ROUTES_WITHOUT_AUTH_WRAPPER: ValueOf<typeof AppRoute>[] = [AppRoute.QUIZ];
```

File: apps/frontend/src/libs/components/protected-route/libs/constants/routes-with-auth-wrapper.constant.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import { AppRoute } from "~/libs/enums/enums.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/libs/constants/routes-with-auth-wrapper.constant.ts:1]: Change the file name

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Comment by ihorLysak** [comment]
Sorry for non-informative commit name, i pushed it before the daily meeting

**Review by yulyaolshanska — APPROVED** [review]

File: apps/frontend/src/libs/components/protected-route/libs/constants/routes-without-auth-wrapper.constant.ts [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import { AppRoute } from "~/libs/enums/enums.js";
+import { type ValueOf } from "~/libs/types/types.js";
+
+const ROUTES_WITHOUT_AUTH_WRAPPER = [AppRoute.QUIZ] as ValueOf<
+	typeof AppRoute
+>[];
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/libs/constants/routes-without-auth-wrapper.constant.ts:6]:
```suggestion
const ROUTES_WITHOUT_AUTH_WRAPPER: ValueOf<typeof AppRoute>[] = [AppRoute.QUIZ];
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Comment by ihorLysak** [comment]
I accidentaly made work intended for another branch, those changes are reverted now

**Review by fshabanov — APPROVED** [review]

**Pull request merged**